server/logging_test.go

```diff
@@ -0,0 +1,184 @@
+		err := json.Unmarshal(targetCfg.Options, logOptionsConsole)
+		require.NoError(t, err)
+		require.NotNil(t, logOptionsConsole)
+		assert.Equal(t, filename, (*logOptionsConsole)["filename"])
+	})
+}
+
+func TestDefaultLoggingConfigOK(t *testing.T) {
+	t.Run("Console config is get correctly", func(t *testing.T) {
+		cfg := &ConfigPushProxy{
+			EnableFileLog: false,
+		}
+		assert.Equal(t, defaultLoggingConsoleLogConfig(), defaultLoggingConfig(cfg))
+	})
+
+	t.Run("File config is get correctly", func(t *testing.T) {
+		cfg := &ConfigPushProxy{
+			EnableFileLog:   true,
+			LogFileLocation: randomString(10),
+		}
+		assert.Equal(t, defaultLoggingFileLogConfig(cfg.LogFileLocation), defaultLoggingConfig(cfg))
+	})
+}
+
+func TestNewMlogLoggerConsoleLegacyOK(t *testing.T) {
+	t.Run("Instancing logger with console for legacy conf", func(t *testing.T) {
+		cfg := &ConfigPushProxy{
+			EnableFileLog: false,
+		}
+		logger, err := NewMlogLogger(cfg)
+		assert.NoError(t, err)
+		assert.NotNil(t, logger)
+	})
+}
+
+func TestNewMlogLoggerFileLegacyOk(t *testing.T) {
+	t.Run("Instancing logger with file for legacy conf", func(t *testing.T) {
+		log, err := os.CreateTemp("", "log")
+		require.NoError(t, err)
+
+		err = log.Close()
+		require.NoError(t, err)
+		defer os.Remove(log.Name())
+
+		cfg := &ConfigPushProxy{
+			EnableFileLog:   true,
+			LogFileLocation: log.Name(),
+		}
+
+		logger, err := NewMlogLogger(cfg)
+		assert.NoError(t, err)
+		assert.NotNil(t, logger)
+	})
+}
+
+func TestNewMlogLoggerLoggingCfgFileOk(t *testing.T) {
+	t.Run("Instancing logger with file", func(t *testing.T) {
+		conf, err := os.CreateTemp("", "logget-cfg-conf.json")
+		require.NoError(t, err)
+
+		_, err = conf.WriteString(defaultLoggingConsoleLogConfig())
+		require.NoError(t, err)
+
+		err = conf.Close()
+		require.NoError(t, err)
+		defer os.Remove(conf.Name())
+
+		cfg := &ConfigPushProxy{
+			LoggingCfgFile: conf.Name(),
+		}
+
+		logger, err := NewMlogLogger(cfg)
+		assert.NoError(t, err)
+		assert.NotNil(t, logger)
+	})
+}
+
+func TestNewMlogLoggerLoggingCfgJSONOk(t *testing.T) {
+	t.Run("Instancing logger with json", func(t *testing.T) {
+		conf, err := os.CreateTemp("", "logget-cfg-conf.json")
+		require.NoError(t, err)
+
+		_, err = conf.WriteString(defaultLoggingConsoleLogConfig())
+		require.NoError(t, err)
+
+		err = conf.Close()
+		require.NoError(t, err)
+		defer os.Remove(conf.Name())
+
+		cfg := &ConfigPushProxy{
+			LoggingCfgJSON: defaultLoggingConsoleLogConfig(),
+		}
+
+		logger, err := NewMlogLogger(cfg)
+		assert.NoError(t, err)
+		assert.NotNil(t, logger)
+	})
+}
+
+func randomString(length int) string {
+	// Define the character set
+	const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789\""
+	seededRand := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	// Generate the string
+	result := make([]byte, length)
+	for i := range result {
+		result[i] = charset[seededRand.Intn(len(charset))]
+	}
+	return string(result)
+}
```

Comment:
This may not be needed, since (I think) you can use `model.NewId()`, which will give you a random string.